apps/guardian-ui/src/GuardianApi.ts

```diff
@@ -366,8 +371,20 @@ export class GuardianApi
     return this.base.call(AdminRpc.version);
   };
 
-  fetchBlockCount = (): Promise<number> => {
-    return this.base.call(AdminRpc.fetchBlockCount);
+  fetchBlockCount = (config: ClientConfig): Promise<number> => {
+    const walletModuleId = config
+      ? Object.entries(config.modules).find(
+          (m) => m[1].kind === ModuleKind.Wallet
+        )?.[0]
+      : undefined;
+
+    if (!walletModuleId) {
+      throw new Error('No wallet module found');
+    }
+    return this.moduleApiCall(
+      Number(walletModuleId),
+      WalletModuleRpc.blockCount
+    );
```

Comment:
For module api calls I think it's better to pass the config in as an argument like this in order to find the instance ID. https://github.com/fedimint/ui/blob/3291e5b01e10ac46bd65497f79068079cb3f409c/apps/guardian-ui/src/components/GatewaysCard.tsx#L35-L50 has this sort of logic in the view layer which doesn't seem right.